EngineBay.Authentication/Cookies/SignIn.cs

```diff
@@ -0,0 +1,51 @@
+namespace EngineBay.Authentication.Cookies
+{
+    using System.Security.Claims;
+    using EngineBay.Core;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.AspNetCore.Authentication.Cookies;
+
+    public class SignIn : ICommandHandler<SignInCredentials, bool>
+    {
+        private readonly AuthenticationDbContext authDb;
+        private readonly VerifyBasicAuthCredentials verifyBasicAuthCredentials;
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        public SignIn(IHttpContextAccessor httpContextAccessor, AuthenticationDbContext authDb, VerifyBasicAuthCredentials verifyBasicAuthCredentials)
+        {
+            this.httpContextAccessor = httpContextAccessor;
+            this.authDb = authDb;
+            this.verifyBasicAuthCredentials = verifyBasicAuthCredentials;
+        }
+
+        public async Task<bool> Handle(SignInCredentials command, CancellationToken cancellation)
+        {
+            ArgumentNullException.ThrowIfNull(command);
+            ArgumentNullException.ThrowIfNull(this.httpContextAccessor);
+            ArgumentNullException.ThrowIfNull(this.httpContextAccessor.HttpContext);
+
+            var claims = new[] { new Claim(CustomClaimTypes.Name, command.Username) };
+            var claimsIdentity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);
+            var claimsPrincipal = new ClaimsPrincipal(claimsIdentity);
+
+            var basicAuthCredentialsDto = new BasicAuthCredentialsDto()
+            {
+                ClaimsPrincipal = claimsPrincipal,
+                Password = command.Password,
+            };
+
+            var authProperties = new AuthenticationProperties
+            {
```

Comment:
Is this meant to be empty?

Reply:
for now yes.
we can set things like refresh and expiry, so I don't want to remove it yet before exploring those options